tests/integration/SdkTest_test.cpp

```diff
@@ -209,69 +209,48 @@ namespace
     }
 }
 
-std::map<int, std::string> gSessionIDs;
-
-void SdkTest::SetUp()
+std::string logTime()
 {
-    // do some initialization
-    if (megaApi.size() < 2)
+    // why do the tests take so long to run?  Log some info about what is slow.
+    auto t = std::time(NULL);
+    char ts[50];
+    struct tm dt;
+    ::mega::m_gmtime(t, &dt);
+    if (!std::strftime(ts, sizeof(ts), "%H:%M:%S ", &dt))
     {
-        megaApi.resize(2);
-        mApi.resize(2);
+        ts[0] = '\0';
     }
-    char *buf = getenv("MEGA_EMAIL");
-    if (buf)
-        mApi[0].email.assign(buf);
-    ASSERT_LT((size_t)0, mApi[0].email.length()) << "Set your username at the environment variable $MEGA_EMAIL";
+    return ts;
+}
 
-    buf = getenv("MEGA_PWD");
-    if (buf)
-        mApi[0].pwd.assign(buf);
-    ASSERT_LT((size_t)0, mApi[0].pwd.length()) << "Set your password at the environment variable $MEGA_PWD";
+std::map<int, std::string> gSessionIDs;
 
+void SdkTest::SetUp()
+{
     gTestingInvalidArgs = false;
-
-    if (megaApi[0].get() == NULL)
-    {
-        megaApi[0].reset(new MegaApi(APP_KEY.c_str(), megaApiCacheFolder(0).c_str(), USER_AGENT.c_str(), int(0), unsigned(THREADS_PER_MEGACLIENT)));
-        mApi[0].megaApi = megaApi[0].get();
-
-        megaApi[0]->setLoggingName("0");
-        megaApi[0]->addListener(this);
-
-        LOG_info << "___ Initializing test (SetUp()) ___";
-
-        if (!gResumeSessions || gSessionIDs[0].empty())
-        {
-            ASSERT_NO_FATAL_FAILURE( login(0) );
-        }
-        else
-        {
-            ASSERT_NO_FATAL_FAILURE( loginBySessionId(0, gSessionIDs[0].c_str()) );
-        }
-
-        ASSERT_NO_FATAL_FAILURE( fetchnodes(0) );
-    }
-
-    // In case the last test exited without cleaning up (eg, debugging etc)
-    Cleanup();
 }
 
 void SdkTest::TearDown()
 {
+    cout << logTime() << "Test done, teardown starts" << endl;
     // do some cleanup
 
-    if (gResumeSessions && gSessionIDs[0].empty())
+    for (int i = 0; i < gSessionIDs.size(); ++i)
     {
-        if (auto p = unique_ptr<char[]>(megaApi[0]->dumpSession()))
+        if (gResumeSessions && megaApi[i] && gSessionIDs[i].empty())
         {
-            gSessionIDs[0] = p.get();
+            if (auto p = unique_ptr<char[]>(megaApi[i]->dumpSession()))
+            {
+                gSessionIDs[i] = p.get();
+            }
         }
     }
 
     gTestingInvalidArgs = false;
 
     LOG_info << "___ Cleaning up test (TearDown()) ___";
+
+    cout << logTime() << "Cleaning up account" << endl;
```

Comment:
I find the output extremely useful, myself, and in this instance I was trying to find out why the tests are so slow to run, especially from my own machine - for example here I can see that Jenkins can login in approximately 0 seconds - it takes 11 typically from my machine.   And fetchnodes is taking 35 seconds, that is disappointingly long, maybe these test accounts have too much history, especially the Jenkins ones.  Waiting for the tests to run is a factor in how fast we can develop, I think.     How about I change this one so we can turn that cout output on or off with a command line switch, so we can keep everyone happy :-)  cheers

Reply:
What if we use the log rather than `cout` for this purpose? You know the messages you're looking for, so checking the log rather than the standard output should be fine?
Otherwise, a way to enable/disable those messages should be ok (please, disabled by default) :)

Reply:
Yes I've added that with those lines disabled by default.   I have tried monitoring the log, but there is so much output there, you can't see what is going on as those lines (mostly low level networking related) scroll by frantically - looking at the test as it runs, that is.   With the cout lines disabled, it can take minutes for a single test to run, and it's nice to have an idea about what it's working on at that moment, and know that it's not stuck on some much earlier step, in which case waiting for the test to end - and it may be a long time - is wasted time.

Reply:
Ah I see I had not converted the Sync_test.cpp to out() - I hadn't worried about that as those are not run in Jenkins (yet - maybe we could add them now though?) - anyway, that file is now converted also, thanks

Reply:
Thanks, Matt. Ready to go!